ksqldb-engine/src/test/java/io/confluent/ksql/engine/rewrite/StatementRewriterTest.java

```diff
@@ -170,7 +170,8 @@ private Query givenQuery(
       final Optional<Expression> where,
       final Optional<GroupBy> groupBy,
       final Optional<PartitionBy> partitionBy,
-      final Optional<Expression> having
+      final Optional<Expression> having,
+      final Optional<ResultMaterialization> resultMaterialization
```

Comment:
Sorry should we not be add `resultMaterialization` parameter in this method then? The tests did fail as you said but I changed the tests to expect `Optional.empty()` similar to the other parameters and they now pass

Reply:
we should make sure to test it both ways. I think it makes sense to pass it in here but then we need to test that it works as well. If you look at the tests in this class they all follow a pattern:
```
// Given
query := a query
mock(expressionRewriter).toDoSomething

// When
rewriter.rewrite(query)

// Then
assertThat(toDoSomething happened)
```
Similarly, we should add a test to make sure that if the mock rewriter rewrites the resultMaterialization that we get the right thing. In most cases, the `givenQuery` should take in `Optional.empty()` (and expect the output to be empty) but we should have one test where it expects the materialization to be rewritten

Reply:
So actually looking closer I see that `StatementRewriter` only works on `AstNodes` or `expressions`, and so cannot actually rewrite `resultMaterialization` since it does not extend either of these.  So some options are we could leave the test as if for now, or revert the test back to when it just used the field, or we could add a new function to rewrite the `resultMaterialization` though I am not sure how important it is right now to get it rewritten. I am leaning towards option 1 or 2 but want to see what your thoughts are.

Reply:
I don't imagine we're going to be rewriting it anytime soon, so I'm happy to revert the test back to just having it be a field